src/blueprint/from_config/operators/http.rs

```diff
@@ -30,6 +30,32 @@ impl<'a> MustachePartsValidator<'a> {
   fn new(type_of: &'a config::Type, config: &'a Config, field: &'a FieldDefinition) -> Self {
     Self { type_of, config, field }
   }
+  fn get_nested_type(&self, tail: &[String], mut type_of: &'a config::Type) -> Result<Type, String> {
+    let mut len = tail.len();
+    for item in tail {
+      if let Some(val_type) = get_value_type(type_of, item) {
+        match len {
+          1 => {
+            return Ok(val_type);
+          }
+          _ => {
+            if val_type.is_nullable() {
+              return Err(format!("value '{}' is a nullable type", item.as_str()));
+            }
+          }
+        }
+      } else {
+        return Err(format!("no value '{}' found", item));
+      }
+      let field = type_of.fields.get(item).ok_or(format!("no value '{}' found", item))?;
```

Comment:
`ok_or` part is basically already checked in the `get_value_type`. And as `get_value_type` is used only in this function we can change it or remove to simplify that function and to not to duplicate the logic

Reply:
I've removed the `ok_or`, but I'll let `get_value_type` as it is to avoid more nested conditions. if you prefer to leave this logic on this method, let me know, and I can delete `get_value_type`.